### PR TITLE
Core Lib Documentation: `Syscalls` module 

### PR DESCRIPTION
**Pull request by TAdev0**

### CODE REVIEW

**Review by enitrat — CHANGES_REQUESTED** [review]
There is too much information missing, not enough context on how to use these functions, their side effects, whether they should actually be used directly or not, etc

I see you're taken some elements from https://docs.starknet.io/architecture-and-concepts/smart-contracts/system-calls-cairo1/#get_execution_info but you left out most of the interesting details.


Focus on the "how", "why", not only the "what"

File: corelib/src/starknet/syscalls.cairo [hunk 1]
```diff
@@ -28,88 +48,125 @@ pub extern fn deploy_syscall(
...
     address_domain: u32, address: StorageAddress, value: felt252,
 ) -> SyscallResult<()> implicits(GasBuiltin, System) nopanic;
 
-
 /// Replaces the class hash of the current contract.
-/// `class_hash` - The class hash that should replace the current one.
+///
+/// # Arguments
+///
+/// * `class_hash` - The class hash that should replace the current one.
```
[the remark this answers is not in the document]
Comment (TAdev0) [reply corelib/src/starknet/syscalls.cairo:135]: yes

[the remark this answers is not in the document]
Comment (TAdev0) [reply corelib/src/starknet/syscalls.cairo:135]: see new cairo campaign, middle floor part 2 :p

File: corelib/src/starknet/syscalls.cairo [hunk 2]
```diff
@@ -28,88 +48,125 @@ pub extern fn deploy_syscall(
...
 pub extern fn get_class_hash_at_syscall(
     contract_address: ContractAddress,
 ) -> SyscallResult<ClassHash> implicits(GasBuiltin, System) nopanic;
 
-
 /// Computes the keccak of the input.
+///
+/// # Arguments
+///
+/// * `input` - The input provided to the keccak function.
```
[the remark this answers is not in the document]
Comment (TAdev0) [reply corelib/src/starknet/syscalls.cairo:155]: no. and this is already clear:

```
/// The system call does not add any padding and the input needs to be a multiple of 1088 bits
/// (== 17 u64 word).
```

File: corelib/src/starknet/syscalls.cairo [hunk 3]
```diff
@@ -28,88 +48,125 @@ pub extern fn deploy_syscall(
...
+/// complete description of this information, see [`Execution information`].
+///
+/// Returns a box containing the current execution information.
 pub extern fn get_execution_info_syscall() -> SyscallResult<
     Box<starknet::info::ExecutionInfo>,
 > implicits(GasBuiltin, System) nopanic;
 
 /// Gets information about the current execution, version 2.
+///
+/// Returns a box containing the current V2 execution information.
```
[the remark this answers is not in the document]
Comment (TAdev0) [reply corelib/src/starknet/syscalls.cairo:79]: indeed this syscall is better not used directly there are wrappers for it

File: corelib/src/starknet/syscalls.cairo (hunk 1 above)
[the remark this answers is not in the document]
Comment (enitrat) [reply corelib/src/starknet/syscalls.cairo:148]: > yes

well, the correct answer is actually no.

consider a contract with

```
// Original Class
fn upgrade(ref self: ContractState){
   replace_class(...);
   println!("hello")
}

// New Class
fn upgrade(ref self: ContractState){
   replace_class(...);
   println!("hello");
}
```

-> The new class is _only effective_ after the end of the current call.

so to execute new code, you'd need to return from the current call and make another call

**Review by enitrat — APPROVED** [review]

**Pull request closed without merging**